### PR TITLE
Feat(networks): add support for network switch and allow

### PR DESCRIPTION
**Pull request by 0xGabi**
Add support for [EIP-3326](https://eips.ethereum.org/EIPS/eip-3326) and [EIP-3085](https://eips.ethereum.org/EIPS/eip-3085).

### CODE REVIEW

File: src/networks.js [hunk 1]
```diff
@@ -65,17 +57,69 @@ const networks = {
...
+      method: 'wallet_switchEthereumChain',
+      params: [{ chainId: chainIdHex }],
+    })
+  } catch (switchError) {
+    // This error code indicates that the chain has not been added to Injected provider.
+    if (switchError.code === 4902) {
+      try {
+        await addEthereumChain()
+      } catch (addError) {
+        // handle "add" error
```
Comment (rperez89) [inline src/networks.js:115]: maybe here at least we can throw a console.error for now

File: src/networks.js [hunk 2]
```diff
@@ -65,17 +57,69 @@ const networks = {
...
+      }
+    }
+    // handle other "switch" errors
+    // TODO: handle errors
```
Comment (rperez89) [inline src/networks.js:119]: same here

**Review by rperez89 — COMMENTED** [review]
Awesome work gabi!!

**Comment by 0xGabi** [comment]
I address the logs in #672

**Pull request merged**